lang/java/avro/src/main/java/org/apache/avro/io/parsing/ResolvingGrammarGenerator.java

```diff
@@ -294,7 +296,14 @@ static void encode(Encoder e, Schema s, JsonNode n) throws IOException {
         throw new AvroTypeException("Non-string default value for fixed: " + n);
       byte[] bb = n.textValue().getBytes(StandardCharsets.ISO_8859_1);
       if (bb.length != s.getFixedSize()) {
-        bb = Arrays.copyOf(bb, s.getFixedSize());
+        int offset = s.getFixedSize() - bb.length;
+        byte signum = (byte) (new BigDecimal(new BigInteger(bb)).signum() < 0 ? 0xFF : 0x00);
+        byte[] fixedBytes = new byte[s.getFixedSize()];
+
+        // Fill the front of the array and copy remaining with unscaled values
+        Arrays.fill(fixedBytes, 0, offset, signum);
```

Comment:
Offset can be negative if the default is too long, which gives an IllegalArgumentException.  The previous behaviour was to truncate it.

Reply:
Hello, according to the original code, if the fixed size is larger than the default value, it needs to be filled. If it is smaller than the default value, an exception will be thrown, but I am not sure whether to give up this behavior in 1.9.x

Reply:
And if the sizes don't match, an exception is thrown, which I don't think is friendly, unless the problem cannot be solved

Reply:
> If it is smaller than the default value, an exception will be thrown

Hmmm -- after a quick check, if the default value is longer than the fixed size, the original code silently trims the default value without an exception.